SPECS/python-sphinx/python-sphinx.spec

```diff
@@ -2,212 +2,165 @@
-    * Automatic indices: general index as well as a module index
-    * Code handling: automatic highlighting using the Pygments highlighter
-    * Various extensions are available, e.g. for automatic testing of
-      snippets and inclusion of appropriately formatted docstrings.
+strength come from the power and straightforwardness of reStructuredText
+and its parsing and translating suite, the Docutils.
 
-%prep
-%autosetup -n Sphinx-%{version} -p1
+%package doc
+Summary:        Documentation for %{name}
+License:        BSD-2-Clause
+Recommends:     python%{python3_pkgversion}-sphinx = %{version}-%{release}
 
-# Remove bundled JS components.
-rm ./sphinx/themes/basic/static/{jquery,underscore}*.js
-rm ./sphinx/themes/bizstyle/static/css3-mediaqueries*.js
+%description doc
+Sphinx makes it easy to create intelligent and beautiful documentation.
 
-%if %{without imagemagick_tests}
-rm tests/test_ext_imgconverter.py
-%endif
+Sphinx uses reStructuredText as its markup language, and many of its
+strength come from the power and straightforwardness of reStructuredText
+and its parsing and translating suite, the Docutils.
 
-# Don't measure coverage:
-sed -i '/pytest-cov/d' setup.py
-# Not needed on recent Pythons, https://github.com/sphinx-doc/sphinx/pull/8483
-sed -i '/typed_ast/d' setup.py
+This package contains documentation in the HTML format.
 
+%prep
+%autosetup -n sphinx-%{version} -p1
 
-%generate_buildrequires
-%pyproject_buildrequires -r %{?with_check:-x test}
+# We don't have ImageMagick at the moment
+rm tests/test_ext_imgconverter.py
+
+# Sphinx' tests import from each other, this feature is not supported by
+# the 'importlib' import mode in pytest. Upstream mitigates this by invoking
+# `python -m pytest` rather than `pytest` directly, but in the context of the
+# RPM build we explicitly want to test the installed library rather than the
+# one from PWD.
+# https://github.com/sphinx-doc/sphinx/issues/11740
+sed -i '/"--import-mode=importlib",/d' pyproject.toml
 
 %build
 %pyproject_wheel
 
+export PYTHONPATH=$PWD
+pushd doc
+export SPHINXBUILD="%{python3} ../sphinx/cmd/build.py"
+make html SPHINXBUILD="$SPHINXBUILD"
+make man SPHINXBUILD="$SPHINXBUILD"
+rm -rf _build/html/.buildinfo
+# Those files are copied to _build/html/_images and loaded to the
+# html pages from there - we can safely remove the duplicated and unused files
+rm -rf _build/html/_static/themes _build/html/_static/tutorial
+rm -f _build/html/_static/more.png _build/html/_static/translation.svg
+mv _build/html ..
+popd
+
 %install
 %pyproject_install
 
 # For backwards compatibility. Remove with care, if at all
 for i in sphinx-{apidoc,autogen,build,quickstart}; do
-    ln -s %{_bindir}/$i %{buildroot}%{_bindir}/${i}%{python3_pkgversion}
     ln -s %{_bindir}/$i %{buildroot}%{_bindir}/$i-%{python3_version}
-    ln -s %{_bindir}/$i %{buildroot}%{_bindir}/$i-%{python3_pkgversion}
+    ln -s %{_bindir}/$i %{buildroot}%{_bindir}/$i-3
```

Comment:
Should this be hardcoded to 3? I see fedora has it hardcoded but it seems odd.

Reply:
hmm I think it's acceptable either way, but I'll change it back to using macro just so the meaning is clearer.